From f8ee363ec13a74eeb4badc0aedc57a9ad8c9c13f Mon Sep 17 00:00:00 2001
From: He Zhe <zhe.he@windriver.com>
Date: Mon, 8 Aug 2022 07:58:55 +0000
Subject: [PATCH] sched: deadline: Use printk_once instead of missing
 printk_deferred_once

This is a temporary fix which will be replaced by upstream update later.

46d5575df8a1 ("sched/deadline: Fix BUG_ON condition for deboosted tasks")
uses printk_deferred_once while
25f13bd1d07b ("printk: remove deferred printing") remove the function in
RT upstream. Let's use printk_once instead to fix the build failure.

kernel-source/kernel/sched/deadline.c: In function 'enqueue_task_dl':
kernel-source/kernel/sched/deadline.c:1565:25: error:
implicit declaration of function 'printk_deferred_once' [-Werror=implicit-function-declaration]
1565 |                         printk_deferred_once("sched: DL de-boosted task PID %d: REPLENISH flag missing\n",
     |                         ^~~~~~~~~~~~~~~~~~~~

Upstream-Status: Pending

Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 kernel/sched/deadline.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/deadline.c b/kernel/sched/deadline.c
index c9dc1bc06431..4c6579cb1b7f 100644
--- a/kernel/sched/deadline.c
+++ b/kernel/sched/deadline.c
@@ -1562,7 +1562,7 @@ static void enqueue_task_dl(struct rq *rq, struct task_struct *p, int flags)
 		 */
 		p->dl.dl_throttled = 0;
 		if (!(flags & ENQUEUE_REPLENISH))
-			printk_deferred_once("sched: DL de-boosted task PID %d: REPLENISH flag missing\n",
+			printk_once("sched: DL de-boosted task PID %d: REPLENISH flag missing\n",
 					     task_pid_nr(p));
 
 		return;
-- 
2.32.0

